server/grpc_bytestream.go

```diff
@@ -129,26 +129,16 @@ func (s *grpcServer) Read(req *bytestream.ReadRequest,
 	var chunkResp bytestream.ReadResponse
 	for {
 		n, err := rdr.Read(buf)
-		eof := err == io.EOF
-		if eof {
-			err = nil
-		}
 
-		if err != nil {
-			msg := fmt.Sprintf("GRPC BYTESTREAM READ FAILED: %v", err)
-			s.accessLogger.Printf(msg)
-			return status.Error(codes.Unknown, msg)
-		}
-
-		if limitedSend {
-			if (sendLimitRemaining - int64(n)) < 0 {
-				msg := "GRPC BYTESTREAM READ FAILED: READ LIMIT EXCEEDED"
-				return status.Error(codes.OutOfRange, msg)
+		if n > 0 {
+			if limitedSend {
+				if (sendLimitRemaining - int64(n)) < 0 {
+					msg := "GRPC BYTESTREAM READ FAILED: READ LIMIT EXCEEDED"
+					return status.Error(codes.OutOfRange, msg)
+				}
+				sendLimitRemaining -= int64(n)
 			}
-			sendLimitRemaining -= int64(n)
-		}
 
-		if n > 0 {
 			chunkResp.Data = buf[:n]
 			err = resp.Send(&chunkResp)
```

Comment:
I think we need to use a different variable here, otherwise this will overwrite the value from `Read`.

Reply:
ah yes, fixed.